### ISSUE TITLE
Daemon won't start

### ISSUE THREAD

**Issue by lattice0** [openrazer/openrazer#446]
```
Traceback (most recent call last):
  File "/usr/bin/openrazer-daemon", line 168, in <module>
    run()
  File "/usr/bin/openrazer-daemon", line 163, in run
    logger=logger)
TypeError: __init__() got an unexpected keyword argument 'chdir'
```

Did

`    print(args.run_dir)`

and got

`/run/user/1000`

**Comment by lattice0** [comment]
If I take 

`                       foreground=args.foreground,`
and
`                       chdir=args.run_dir,`

out the daemon runs, but I get no devices found (which might be a different problem). Anyways, I compiled everything myself and installed. Do I need to do something more?

**Comment by lattice0** [comment]
If I take 

`                       foreground=args.foreground,`
and
`                       chdir=args.run_dir,`

out the daemon runs, but I get no devices found (which might be a different problem). Anyways, I compiled everything myself and installed. Do I need to do something more?

**Comment by lattice0** [comment]
update: after reboot it worked

**Comment by z3ntu** [comment]
What distro are you using?

**Comment by lattice0** [comment]
Ubuntu 17.04.03

-------- Original Message --------

**Comment by z3ntu** [comment]
So it works now or do you still need help?

**Comment by lattice0** [comment]
it works but I think it did only because I restarted. Maybe someone will have this bug too.

-------- Original Message --------

**Comment by z3ntu** [comment]
So we can close this?

**Comment by lattice0** [comment]
yes

**Comment by jamieei** [comment]
Same problem on debian stable (stretch). I think that the problem is that I have version 2.3.1-1 of python3-daemonize, which is too old to have the chdir and foreground params in __init__.

Not fixed by rebooting.

**Comment by z3ntu** [comment]
@JamieEi does `apt upgrade` upgrade the package?

**Comment by jamieei** [comment]
No, there's not a newer version in stable or backports. Only unstable.

**Comment by z3ntu** [comment]
I mean in the repository you have installed the openrazer packages from.

**Comment by jamieei** [comment]
No, I get openrazer from
http://download.opensuse.org/repositories/hardware:/razer/Debian_Sid/.
python3-daemonize
comes from the Debian repos. Sid (unstable) has a recent version with the
missing __init__ flags but Stretch (stable) does not, including backports.

**Comment by z3ntu** [comment]
Why did you use the Debian_Sid repo then instead of the Debian_9.0 repo if you use Debian 9?

**Comment by jamieei** [comment]
I didn't. I only use stable and backports. Sid has a newer package that
would presumably work, but I don't want to mix them.

**Comment by z3ntu** [comment]
You wrote in the last comment that you used http://download.opensuse.org/repositories/hardware:/razer/Debian_Sid/

**Comment by jamieei** [comment]
Ah sorry, I must have grabbed the wrong command! Let me try again.

**Comment by jamieei** [comment]
Yep, that was it. Sorry for the confusion.
